src/main/scala/com/fulcrumgenomics/bam/TrimPrimers.scala

```diff
@@ -106,14 +112,26 @@ class TrimPrimers
     //         i) No reference was given (so no coordinate sort) and the output sort order is queryname, or
     //        ii) A reference was given (so the last sort is coordinate) and the output sort order is coordinate
     //      else, we'll have to sort for potentially a third time in the SAMFileWriter
-    val (sorter, write: (SamRecord => Any), out: SamWriter) = ref match {
+    val (sorter, write: (SamRecord => Any), out: SamWriter, write_primerless_opt: Option[(SamRecord => Any)], primerless_out_opt: Option[SamWriter]) = ref match {
```

Comment:
Please use lower camel case for variables and methods throughout.  See: https://docs.scala-lang.org/style/naming-conventions.html

Reply:
This is getting quite difficult to read (not your fault), let me make PR to clean this up